### ISSUE TITLE
Support Multiple Types in AlterConfig Struct?

### ISSUE THREAD

**Issue by marshmallory** [twmb/franz-go#103]
The AlterConfig struct only accepts type `*string`.  Is it possible to alter this to accept more types, or is this a limitation specific to Kafka where it only takes in the string?

In our use case, we're looking at accepting multiple types from the user's YAML, and parsing it in our kafka-provider to then be applied.  We have been successful getting the native strings to work, but less so with int64s and bools.

Link to the issue:
https://github.com/crossplane-contrib/provider-kafka/issues/5

Happy to provide more details!

**Comment by twmb** [comment]
This is specific to the protocol itself; number need to be converted into strings "10", booleans to "true", etc. I think strconv.Format{Bool,Float,Int}, or `fmt.Sprintf("%v", ...)` suffice for non-null values.

**Comment by twmb** [comment]
Perhaps, though, the `kadm` package's `AlterConfig.Value` can be changed to an interface that supports ints/floats/bools/strings/[]bytes. What do you think?

**Comment by jograca** [comment]
We'll give it a try! Thanks

**Comment by twmb** [comment]
I think I'll switch the value to an `interface{}`, which will make this easier to use.

**Comment by turkenh** [comment]
I feel like having the client use the same underlying type as Kafka protocol would be more intuitive. 

@marshmallory maybe we also take this configuration as key/value pairs in our API to be closer to Kafka API instead of dealing with converting from/to string=>int/bool? 

IIUC, other clients/tools also use this approach to configure Kafka resources which might be an outcome of what @twmb mentions about the protocol.

**Comment by twmb** [comment]
I looked into this a bit -- using an `interface{}` is easier to use for talking, but doesn't clear anything up when trying to handle the result of describing configs. I wouldn't want the client to look at a string and try to guess its type for _you_, which means that you'd have to understand what the config type is in the end anyway.

So, I think it may be worth keeping the current `*string` aspect and having the users format into a string properly, as @turkenh mentions.

**Comment by marshmallory** [comment]
Yes, I think you're both right.  

I worked with @turkenh too @twmb, and we ended up doing it a bit differently.  We're just passing in the strings off the bat rather than fuss with all of the random types and conversions.  It's not quite as good as validating each type would be and then converting it, but it simply wasn't worth the extra fuss just for admin configs that won't be set most of the time (for our users).

Thanks for the conversation!  I'm going to close this now.
